src/Raven.Client/Json/TypeConverter.cs

```diff
@@ -7,108 +7,141 @@
 using Sparrow.Extensions;
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
+using Sparrow.Utils;
 
 namespace Raven.Client.Json
 {
     internal static class TypeConverter
     {
-        public static object ToBlittableSupportedType(object value, DocumentConventions conventions, JsonOperationContext context)
+
+        private enum BlittableSupportedReturnType : int
         {
-            if (value == null)
-                return null;
+            Null = 0,
+            Same = 1,
+            Runtime = 3,
+            Dictionary = 5,
+            GenericDictionary = 6,
+            Enumerable = 7,
+            String = 8
+        }
 
-            if (value is BlittableJsonReaderObject)
-                return value;
+        private static readonly TypeCache<BlittableSupportedReturnType> _supportedTypeCache = new(512);
```

Comment:
What is this class?

Reply:
It's an optimized open hash table for types. Freaking simple and fast.

Reply:
where is the code for that?